crates/pallet-dummy-precompiles-code/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            for precompile_address in &T::PrecompilesAddresses::get() {
+                pallet_evm::Pallet::<T>::create_account(*precompile_address, DUMMY_CODE.to_vec());
+            }
+
+            <LastCreationVersion<T>>::put(CURRENT_CREATION_VERSION);
+            <ForceUpdateAskCounter<T>>::put(T::ForceUpdateAskCounter::get());
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> Weight {
+            let creation_version = Self::creation_version();
+            let last_force_update_ask_counter = Self::force_update_ask_counter();
+            let mut weight = T::DbWeight::get().reads(2);
+
+            if creation_version != CURRENT_CREATION_VERSION {
+                weight += Self::precompiles_addresses_add_dummy_code();
+
+                <LastCreationVersion<T>>::put(CURRENT_CREATION_VERSION);
+                <ForceUpdateAskCounter<T>>::put(T::ForceUpdateAskCounter::get());
+                weight += T::DbWeight::get().writes(1);
+            } else if last_force_update_ask_counter != T::ForceUpdateAskCounter::get() {
+                weight += Self::precompiles_addresses_add_dummy_code();
+
+                <ForceUpdateAskCounter<T>>::put(T::ForceUpdateAskCounter::get());
+                weight += T::DbWeight::get().writes(1);
+            }
+
+            weight
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+            // Do nothing.
+            Ok(Vec::new())
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
+            use sp_std::vec::Vec;
+
+            let mut not_created_precompiles = Vec::new();
+
+            for precompile_address in &T::PrecompilesAddresses::get() {
+                let code = pallet_evm::Pallet::<T>::account_codes(*precompile_address);
+                if code != DUMMY_CODE {
+                    not_created_precompiles.push(*precompile_address);
+                }
+            }
+
+            if !not_created_precompiles.is_empty() {
+                return Err("precompiles not created properly: {:not_created_precompiles}");
+            }
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    /// A helper function to add dummy code for provided precompiles adrresses.
+    fn precompiles_addresses_add_dummy_code() -> Weight {
+        let mut weight = T::DbWeight::get().reads(0);
+
+        for precompile_address in &T::PrecompilesAddresses::get() {
+            let code = pallet_evm::Pallet::<T>::account_codes(*precompile_address);
+            weight += T::DbWeight::get().reads(1);
+
+            if code != DUMMY_CODE {
+                pallet_evm::Pallet::<T>::create_account(*precompile_address, DUMMY_CODE.to_vec());
```

Comment:
Don't we need a system account here too?

Reply:
Which exactly system account are you talking about?

Reply:
`frame_system` / `evm_system` - depending on what's used in `pallet_evm`

Reply:
It's managed under `create_account` fn - https://github.com/humanode-network/frontier/blob/8d3a2ef77867d1e40bd42078eb371436d49f0b1d/frame/evm/src/lib.rs#L705.